ui/src/shared/utils/duration.ts

```diff
@@ -1,11 +1,29 @@
-import {TimeRange} from 'src/types'
+import moment from 'moment'
+
+import {TimeRange, CustomTimeRange} from 'src/types'
 import {Duration, DurationUnit} from 'src/types/ast'
+import {TIME_RANGE_FORMAT} from 'src/shared/constants/timeRanges'
+
+export const removeSpacesAndNow = (input: string): string =>
+  input.replace(/\s/g, '').replace(/now\(\)-/, '')
+
+const durationRegExp = /([0-9]+)(y|mo|w|d|h|ms|s|m|us|µs|ns)/g
+
+export const isDurationParseable = (lower: string): boolean => {
+  if (!lower || !lower.includes('now()')) {
+    return false
+  }
+
+  const removedLower = removeSpacesAndNow(lower)
+
+  return !!removedLower.match(durationRegExp)
+}
 
 export const parseDuration = (input: string): Duration[] => {
-  const r = /([0-9]+)(y|mo|w|d|h|ms|s|m|us|µs|ns)/g
   const result = []
+  const durationRegExp = /([0-9]+)(y|mo|w|d|h|ms|s|m|us|µs|ns)/g
```

Comment:
✂️ this, and use the one defined on line 10

Reply:
Turns out ` As with exec() (or in combination with it), test() called multiple times on the same global regular expression instance will advance past the previous match.`